borg/helpers.py

```diff
@@ -1256,3 +1256,20 @@ def format_time(self, key, item):
 
     def time(self, key, item):
         return safe_timestamp(item.get(key) or item[b'mtime'])
+
+
+COMPRESS_NONE = set("""
+.gz .tgz .bz2 .tbz2 .xz .lzma .lzo .zip .rar .7z
+.mp3 .ogg .oga .flac
+.mp4 .mkv .avi .mpg .mpeg .webm .vob .ts .ogv
+.jpg .jpeg .png .gif
```

Comment:
.mov
.m4v and .m4a
.flv (flash videos)
.aac
.ogm
.dmg (compressed disk images from mac at least), I am sure there are others like this on other platforms.
.txz (I think this is used to shorten .tar.xz)
many popular cameras raw image format are compressed, e.g. canon's .CRW

But overall I suspect it's indeed better to gauge compressibility by some other means as mentioned below. - this will allow to catch all those cases where you cannot know from the extension:
no extension
.exe - might be executable or an sfx archive
.pdf - could be compressed or not
.iso .img - same thing
.tiff - could be uncompressed or compressed (some other image formats too)

Combining both approaches might not be a bad idea too to save on cpu:
i.e. (extensible) blacklist for nocompress, (extensible) whitelist for expensive compress like lzma (for text files and other highly compressible data), autodetect with selected compression for the rest. Also ability to specify different compressions for different path components (lzma,9 for /etc, none for ~/Pictures and ~/Video, zlib for the rest of ~ as an example)